### PR TITLE
Disable notifications

### PR DESCRIPTION
**Pull request by mikeAdamss**
### What

Allow a developer to disable notifications to slack while developing by using the `DISABLE_NOTIFICATIONS` env var. This task: https://github.com/ONSdigital/dp-data-pipelines/issues/49

I've changed the nomanclature to "Notifier" over "Messenger" as its just pass/fail notifications now.

I've created a BasePipelineNotifier base class with two implementations:

- PipelineNotifier - the normal behaviour
- NopNotifier (NOP - non operational). Which has empty methods so will satisfy the code while not actually sending any notifications.

There;'s a function which decides which one you've instantiated based on the `DISABLE_NOTIFICATIONS` env var.

Ran `fmt` and `lint` to tidy up the codebase so there's quite a lot of noise here. To stop this from happening all the time I've added a github action to run everything on push to branches (for now, we should really be using concourse).

### How to review

Try it out. Sanity check.

### Who can review

Anyone.

### CODE REVIEW

File: .github/workflows/code_quality.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+
+      - name: Install pipenv
```
Comment (SarahJohnsonONS) [inline .github/workflows/code_quality.yaml]: `name` should be `Install poetry`?

**Review by SarahJohnsonONS — COMMENTED** [review]
LGTM - just one comment

File: .github/workflows/code_quality.yaml (hunk 1 above)
Comment (mikeAdamss) [reply .github/workflows/code_quality.yaml]: good spot, me and my copy paste

**Review by SarahJohnsonONS — APPROVED** [review]
Approved

**Pull request merged**